core/CronArchive.php

```diff
@@ -964,36 +961,51 @@ private function invalidateWithSegments($idSites, $date, $period, $_forceInvalid
                     $periodObj->getDateTimeEnd()->setTimezone($site->getTimezone())
                 )
             );
-            if ($this->canWeSkipInvalidatingBecauseThereIsAUsablePeriod($params, $doNotIncludeTtlInExistingArchiveCheck)) {
+
+            if ($this->canWeSkipInvalidatingBecauseThereIsAUsablePeriod($params, $skipWhenRunningOrNewEnoughArchiveExists)) {
                 $this->logger->debug('  Found usable archive for {archive}, skipping invalidation.', ['archive' => $params]);
+            } elseif ($skipWhenRunningOrNewEnoughArchiveExists && $this->canWeSkipInvalidatingBecauseInvalidationAlreadyInProgress($site->getId(), $periodObj)) {
+                $this->logger->debug('  Invalidation for {archive} already in progress, skipping invalidation.', ['archive' => $params]);
             } else {
                 $this->getApiToInvalidateArchivedReport()->invalidateArchivedReports(
                     $idSite,
                     $date,
                     $period,
                     $segment = false,
                     $cascadeDown = false,
-                    $_forceInvalidateNonexistent
+                    $period === 'range'
                 );
             }
 
+            $allSegments = $this->segmentArchiving->getAllSegments();
+
             foreach ($this->segmentArchiving->getAllSegmentsToArchive($idSite) as $segmentDefinition) {
-               // check if the segment is available
+                // check if the segment is available
                 if (!$this->isSegmentAvailable($segmentDefinition, [$idSite])) {
                     continue;
                 }
+
+                if ($onlyProcessSegmentsChangedRecently && !$this->wasSegmentChangedRecently($segmentDefinition, $allSegments)) {
```

Comment:
I think we are not yet skipping invalidations aggressively enough, and the archive cascading is intervening :thinking: 

Running `core:archive --skip-segments-today` on a Monday, that is also the second of the month, did not create a day archive for the 2nd. But it created an archive for the week, while archiving logs only showed the month being archived.

Running archiving a second time actually created an invalidation for the week, and processed it, though this time with the expected log messages.

So while we don't create an archive for the single "today" day, and don't create an invalidation for that day, all other periods are still processed without any need.

Modifying [shouldSkipArchiveIfSkippingSegmentArchiveForToday](https://github.com/matomo-org/matomo/blob/9a3ef94df61682e7d543f3f8b438c368f6eb71a6/core/Archive.php#L284) to also skip if the period starts with today may be a start, but I expect it to not be enough :( On the second day of a period (e.g. Tuesday for the week), the week archiving should only run once after the first day of the week has been processed, and then be skipped until the next day, and also not silently be re-archived by another period.